src/util/arith_tools.cpp

```diff
@@ -299,12 +323,36 @@ bool get_bvrep_bit(
 irep_idt
 make_bvrep(const std::size_t width, const std::function<bool(std::size_t)> f)
 {
-  std::string result(width, ' ');
+  std::string result;
+  result.reserve(width / 4 + 1);
```

Comment:
1) If doing the division-by-power-of-two-is-shift then it should be done everywhere (or, my preference: nowhere)
2) `(width + 3) / 4` will avoid reserving an unnecessary extra space.